OpenRA.Game/Map/Map.cs

```diff
@@ -212,89 +61,20 @@ public class Map : IReadOnlyFileSystem
 
-		// Generated data
-		public readonly MapGrid Grid;
-		public IReadOnlyPackage Package { get; private set; }
-		public string Uid { get; private set; }
-
-		public Ruleset Rules { get; private set; }
-		public bool InvalidCustomRules { get; private set; }
-		public Exception InvalidCustomRulesException { get; private set; }
-
-		/// <summary>
-		/// The top-left of the playable area in projected world coordinates
-		/// This is a hacky workaround for legacy functionality.  Do not use for new code.
-		/// </summary>
-		public WPos ProjectedTopLeft { get; private set; }
-
-		/// <summary>
-		/// The bottom-right of the playable area in projected world coordinates
-		/// This is a hacky workaround for legacy functionality.  Do not use for new code.
-		/// </summary>
-		public WPos ProjectedBottomRight { get; private set; }
-
-		public CellLayer<TerrainTile> Tiles { get; private set; }
-		public CellLayer<ResourceTile> Resources { get; private set; }
-		public CellLayer<byte> Height { get; private set; }
-		public CellLayer<byte> Ramp { get; private set; }
-		public CellLayer<byte> CustomTerrain { get; private set; }
-
-		public PPos[] ProjectedCells { get; private set; }
-		public CellRegion AllCells { get; private set; }
-		public List<CPos> AllEdgeCells { get; private set; }
-
-		public event Action<CPos> CellProjectionChanged;
+		public IReadOnlyPackage Package { get; protected set; }
+		public string Uid { get; protected set; }
+		public bool InvalidCustomRules { get; protected set; }
+		public Exception InvalidCustomRulesException { get; protected set; }
 
 		// Internal data
-		readonly ModData modData;
-		CellLayer<short> cachedTerrainIndexes;
-		bool initializedCellProjection;
-		CellLayer<PPos[]> cellProjection;
-		CellLayer<List<MPos>> inverseCellProjection;
-		CellLayer<byte> projectedHeight;
-		Rectangle projectionSafeBounds;
+#pragma warning disable IDE1006 // Naming Styles
+		protected readonly ModData modData;
+		protected Translation Translation;
+#pragma warning restore IDE1006 // Naming Styles
 
-		internal Translation Translation;
+		public abstract void Save(IReadWritePackage package);
 
-		public static string ComputeUID(IReadOnlyPackage package)
-		{
-			return ComputeUID(package, GetMapFormat(package));
-		}
-
-		static string ComputeUID(IReadOnlyPackage package, int format)
-		{
-			// UID is calculated by taking an SHA1 of the yaml and binary data
-			var requiredFiles = new[] { "map.yaml", "map.bin" };
-			var contents = package.Contents.ToList();
-			foreach (var required in requiredFiles)
-				if (!contents.Contains(required))
-					throw new FileNotFoundException($"Required file {required} not present in this map");
-
-			var streams = new List<Stream>();
-			try
-			{
-				foreach (var filename in contents)
-					if (filename.EndsWith(".yaml") || filename.EndsWith(".bin") || filename.EndsWith(".lua") || (format >= 12 && filename == "map.png"))
```

Comment:
This is never added back

Reply:
see #19891

Reply:
Have now added this back in as well as the other linked changes needed from that PR.